DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetAccountLoginWindow.cs

```diff
@@ -0,0 +1,183 @@
+using Rampastring.XNAUI.XNAControls;
+using System;
+using System.Diagnostics;
+using System.Net;
+
+namespace DTAClient.DXGUI.Multiplayer.CnCNet
+{
+    public class CnCNetAccountLoginWindow : XNAWindow
+    {
+        public event Action<object> Cancel;
+        public event Action<bool> LoginSuccess;
+
+        private XNALabel lblPlayerEmail;
+        private XNATextBox tbPlayerEmail;
+        private XNALabel lblPlayerPassword;
+        private XNAPasswordBox tbPlayerPassword;
+        private XNALabel lblLoginWindowTitle;
+        private XNALabel lblError;
+
+        public CnCNetAccountLoginWindow(WindowManager windowManager) : base(windowManager)
+        {
+        }
+        
+        public override void Initialize()
+        {
+            Name = "CnCNetAccountLoginWindow";
+            BackgroundTexture = AssetLoader.LoadTextureUncached("logindialogbg.png");
+            ClientRectangle = new Rectangle(0, 0, 350, 200);
+
+            lblLoginWindowTitle = new XNALabel(WindowManager);
+            lblLoginWindowTitle.Name = "lblWindowTitle";
+            lblLoginWindowTitle.FontIndex = 1;
+            lblLoginWindowTitle.Text = "LOGIN TO CNCNET";
+
+            AddChild(lblLoginWindowTitle);
+            lblLoginWindowTitle.CenterOnParent();
+            lblLoginWindowTitle.ClientRectangle = new Rectangle(
+                lblLoginWindowTitle.X, 12,
+                lblLoginWindowTitle.Width,
+                lblLoginWindowTitle.Height);
+
+            var btnLogin = new XNAClientButton(WindowManager);
+            btnLogin.Name = "btnLogin";
+            btnLogin.ClientRectangle = new Rectangle(12, ClientRectangle.Bottom - 35, 92, 23);
+            btnLogin.Text = "Login";
+            btnLogin.LeftClick += BtnLogin_LeftClick;
+
+            var btnRegister = new XNAClientButton(WindowManager);
+            btnRegister.Name = "btnRegister";
+            btnRegister.ClientRectangle = new Rectangle(btnLogin.Width + 25, ClientRectangle.Bottom - 35, 92, 23);
+            btnRegister.Text = "Register";
+            btnRegister.LeftClick += BtnRegister_LeftClick;
+
+            var btnCancel = new XNAClientButton(WindowManager);
+            btnCancel.Name = "btnCancel";
+            btnCancel.ClientRectangle = new Rectangle(Width - 104,
+                btnLogin.Y, 92, 23);
+            btnCancel.Text = "Cancel";
+            btnCancel.LeftClick += BtnCancel_LeftClick;
+
+            tbPlayerEmail = new XNATextBox(WindowManager);
+            tbPlayerEmail.Name = "tbPlayerEmail";
+            tbPlayerEmail.Text = "email";
+            tbPlayerEmail.ClientRectangle = new Rectangle(100, 50, 200, 19);
+            tbPlayerEmail.Text = "";
+
+            lblPlayerEmail = new XNALabel(WindowManager);
+            lblPlayerEmail.Name = "lblPlayerEmail";
+            lblPlayerEmail.FontIndex = 1;
+            lblPlayerEmail.Text = "Email:";
+            lblPlayerEmail.ClientRectangle = new Rectangle(12, tbPlayerEmail.ClientRectangle.Y + 1,
```

Comment:
Same as above, since you only modify X and Y, use `lblPlayerEmail.X = 12;` and `lblPlayerEmail.Y = tbPlayerEmail.Y + 1` instead of modifying ClientRectangle.